iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp

```diff
@@ -333,18 +335,91 @@ LogicalResult DepthwiseConvOpConversion::matchAndRewrite(
 
   return success();
 }
+
+/// Pattern to convert a linalg.pad_tensor operation into a fill + subtensor
+/// insert. This is needed till pad_tensor op can be fused with its consumers.
```

Comment:
What about fusing pad with its producer ? in this case the producer had to write to the padded layout similar to what we discussed for packing

Reply:
We could do that. Still will need a `subtensor_insert`. Hanhan was looking into it at a point, but ideally though its better to fuse it with its consumers. It should work well with tiling, etc.